### PR TITLE
several features

### PR DESCRIPTION
**Pull request by DavidBJaffe**
- new features
  * complex of changes to support tree generation and a web page describing them
  * single button installation
  * add NALL to turn off all filters
  * add new lead variables nd<k>
  * add option COMPLETE, to remove exact subclonotypes that are missing chains
  * add new lead variable dref = distance from reference outside region of recombination
  * add FCELL to allow e.g. whitelisting or blacklisting of particular barcodes
- improve documentation/usability
  * set default PRE to \~/enclone/datasets,\~/enclone/datasets2 for external runs,
   and make corresponding simplifications to the landing page and the help pages
  * replace "enclone help plot" by an html page that shows examples including pictures
  * restore text about future integration with Cell Ranger and Loupe
  * drop faq item on windows because dated and superceded by page on site
  * update the text on the main enclone help page
  * assert that Ubuntu is preinstalled on most new Windows machines
  * expand ~ and ~user in output file names
  * test for consistency between VDJ and GEX barcodes
- behind the scenes
  * clarify some error messages and add some argument sanity checks
  * speed up a bit by moving some code out of loops
  * some argument processing tidying
  * separate out crate enclone_tail
  * add GUIDE = directory catalog
  * add test for unicode in paths
  * mostly deduplicate crate version numbers using a new "dummy" crate
- for the lvar "mark", now tally on exact subclonotype lines
- fix bug that had crept into e.g. enclone 123085 (internal feature)
- dump "enclone help ideas"
- improve argument handling where SOMETHING= is needed but SOMETHING is given

### CODE REVIEW

**Review by wyattmcdonnell — APPROVED** [review]
Start suggesting, then just tidied with a bunch of tiny commits instead--sorry! Looks great. I will finish testing phylogeny and MSA stuff so we can mint by end of week.

File: pages/tree.html.src [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+<br>
+<img src="../../img/enclone_banner.png" alt="enclone banner" title="enclone banner" width=100% />
+
+<h1>Making phylogenetic trees</h1>
+
+<p>enclone provides several mechanisms for creating and displaying a phylogenetic tree for each 
+clonotype.  These are initial mechanisms, which are likely to be expanded and/or improved over 
+time in response to feedback.  For all of these mechanisms, we recommend using the argument
+<code>COMPLETE</code> to remove exact subclonotypes that are missing some chains.</p>
```
[the remark this answers is not in the document]
Comment (DavidBJaffe) [reply pages/tree.html.src:21]: enclone doesn't issue warnings.  If we think running without COMPLETE is nuts, we can make
it a requirement.
